### PR TITLE
Math magician events

### PR DESCRIPTION
**Pull request by ridaarif98**
- Copy the files `calculator.js` and `operate.js` into a `logic/` directory in my project.
- Import the files in my `Calculator` component.
- Implement the event handlers  to use the math logic from `calculator.js` and `operate.js` in my React component
- Use `state` and `props`, pass props to child components

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  }
+  if (operation === '-') {
+    return one.minus(two).toString();
+  }
+  if (operation === 'x') {
+    return one.times(two).toString();
+  }
+  if (operation === '÷') {
+    return one.div(two).toString();
+  }
```
Comment (Dipeshtwis) [inline src/logic/operate.js]: - [ ]  Please work on the division of a number with zero let us say `9/0` normally you have to get error in the screen but currently you are getting this

Screenshot from 2021-09-15 20-49-51

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  if (operation === '%') {
+    return one.mod(two).toString();
+  }
+  throw Error(`Unknown operation '${operation}'`);
+}
```
Comment (Dipeshtwis) [inline src/logic/operate.js:28]: [OPTIONAL]

- When I tried to do something like `8 +` and then I pressed any operation again like `-`, it's better if it's changed into `8 -` removing the previous operation. But in your case, it's saying invalid which shows in the below screenshot. It's better if you work on that after that your app will be fully bug-free.

Screenshot from 2021-09-15 20-50-11

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @ridaarif98, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: Good use of state and props.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**_Check the comments under the review_**

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Davidosky007** [comment]
## Hi @ridaarif98 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 
nice-job

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**